File: security-management-frontend/src/services/DisponibiliteService.js
import api from "./api";

const DisponibiliteService = {
    getAll: () => api.get("/disponibilites"),
    getById: id => api.get(`/disponibilites/${id}`),
    getByAgent: agentId => api.get(`/disponibilites/agent/${agentId}`),
    create: data => api.post("/disponibilites", data),
    update: (id, data) => api.put(`/disponibilites/${id}`, data),
    delete: id => api.delete(`/disponibilites/${id}`),
};

export default DisponibiliteService;

File: SecurityManagementApp/src/main/java/com/boulevardsecurity/securitymanagementapp/service/DisponibiliteService.java
// src/main/java/com/boulevardsecurity/securitymanagementapp/service/DisponibiliteService.java
package com.boulevardsecurity.securitymanagementapp.service;

import com.boulevardsecurity.securitymanagementapp.dto.DisponibiliteCreationDto;
import com.boulevardsecurity.securitymanagementapp.dto.DisponibiliteDto;

import java.util.List;
import java.util.Optional;

public interface DisponibiliteService {
    List<DisponibiliteDto> getAll();
    Optional<DisponibiliteDto> getById(Long id);
    List<DisponibiliteDto> getByAgent(Long agentId);
    DisponibiliteDto create(DisponibiliteCreationDto creationDto);
    DisponibiliteDto update(Long id, DisponibiliteCreationDto updateDto);
    void delete(Long id);
}

